### ISSUE TITLE
0.13 Support Git Repositories as Terraform provider sources

### ISSUE THREAD

**Issue by JesterOrNot** [hashicorp/terraform-provider-kubernetes-alpha#84]
### Description

I would like to be able to use a git provider as the source for a provider in a `versions.tf` file i.e. the github.com/hashicorp/terraform-provider-kubernetes-alpha provider is not in the registry and it does not work when I put it in either the `.terraform/plugins/registry.terraform.io/kubernetes-alpha` folder as suggested in https://github.com/hashicorp/terraform/blob/guide-v0.13-beta/draft-upgrade-guide.md#explicit-provider-source-locations or in the `.terraform/plugins/linux_amd64` folders citing that it couldn't be installed

### Potential Terraform Configuration

```hcl
terraform {
  required_providers {
    kubernetes-alpha = {
      source = "github.com/hashicorp/terraform-provider-kubernetes-alpha"
    }
  }
  backend "s3" {
    key    = "terraform/dev/terraform_dev.tfstate"
    region = "us-east-2"
    bucket = "aws-tfstate-gitpod"
  }
  required_version = ">= 0.13"
}
```

### References

* https://github.com/hashicorp/terraform-provider-kubernetes-alpha/issues/58
* https://github.com/hashicorp/terraform/blob/guide-v0.13-beta/draft-upgrade-guide.md#explicit-provider-source-locations

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by JesterOrNot** [comment]
Sorry

**Comment by JesterOrNot** [comment]
Wrong repo

**Comment by JesterOrNot** [comment]
Still relevant though how do I work with this module in terraform 0.13

**Comment by aareet** [comment]
@JesterOrNot, that's a good question and we're tracking the progress on the provider being published to the registry in #58. The broader question of how to use custom plugins that aren't in the registry is more related to terraform 0.13 itself, so I'll close this in favor of #58 to avoid duplication.

**Comment by aareet** [comment]
Just to follow up here - https://www.hashicorp.com/blog/automatic-installation-of-third-party-providers-with-terraform-0-13/ has more information on how to set up the path for locally sourced providers. And https://github.com/hashicorp/terraform/issues/25485 resolves some common issues faced when adapting to this new path spec.
